google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/MavenBuildDeploymentSource.java

```diff
@@ -127,17 +115,17 @@ public File getFile() {
         new File(mavenProject.getBuildDirectory()).getPath() + File.separator
             + mavenProject.getFinalName();
 
-    XmlFile appEngineWebXml = AppEngineAssetProvider.getInstance()
-        .loadAppEngineStandardWebXml(project, Collections.singletonList(getModule()));
     AppEngineProjectService projectService = AppEngineProjectService.getInstance();
 
     // The environment will be null for newly deserialized deployment sources to ensure freshness.
     // In this case, we need to reload the environment.
     if (environment == null) {
-      environment = projectService.getModuleAppEngineEnvironment(appEngineWebXml);
+      environment = projectService.getModuleAppEngineEnvironment(getModule()).orElseThrow(
+          () -> new RuntimeException("No environment.")
+      );
     }
 
-    if (environment.isFlexible() && !projectService.isFlexCompat(appEngineWebXml)) {
+    if (environment.isFlexible()) {
```

Comment:
Did you test this out? Flex compat projects should not have this extension appended to them because they are exploded wars. Or is it the case now that compat projects will return true for isFlexible()?